File: android/app/src/main/java/com/meruvakirankumar/memora/model/Memory.kt
package com.meruvakirankumar.memora.model

enum class EventType {
    EXPIRATION,
    PAYMENT,
    RENEWAL,
    RETURN,
    GENERAL;

    fun label(): String = name.lowercase().replaceFirstChar { it.uppercase() }
}

enum class Confidence {
    HIGH,
    MEDIUM,
    LOW;

    fun label(): String = name.lowercase().replaceFirstChar { it.uppercase() }
}

enum class MemoryStatus {
    UPCOMING,
    DUE_TODAY,
    OVERDUE,
    COMPLETED;

    fun label(): String = when (this) {
        UPCOMING -> "Upcoming"
        DUE_TODAY -> "Due Today"
        OVERDUE -> "Overdue"
        COMPLETED -> "Completed"
    }
}

/** A confirmed, stored memory. */
data class Memory(
    val id: String,
    val title: String,
    val eventType: EventType,
    val dateIso: String,
    val confidence: Confidence,
    val sourceText: String,
    val createdAtIso: String,
    val completedAtIso: String? = null,
    val notificationId: Int? = null,
)

/** A memory candidate before the user confirms it. */
data class MemoryCandidate(
    val title: String,
    val eventType: EventType,
    val dateIso: String,
    val confidence: Confidence,
    val sourceText: String,
)

File: ios/Memora/Models/Memory.swift
import Foundation

enum EventType: String, Codable, CaseIterable, Identifiable {
    case expiration = "EXPIRATION"
    case payment = "PAYMENT"
    case renewal = "RENEWAL"
    case returnItem = "RETURN"
    case general = "GENERAL"

    var id: String { rawValue }

    var label: String {
        switch self {
        case .expiration: return "Expiration"
        case .payment: return "Payment"
        case .renewal: return "Renewal"
        case .returnItem: return "Return"
        case .general: return "General"
        }
    }
}

enum Confidence: String, Codable {
    case high = "High"
    case medium = "Medium"
    case low = "Low"

    var label: String { rawValue }
}

enum MemoryStatus {
    case upcoming
    case dueToday
    case overdue
    case completed

    var label: String {
        switch self {
        case .upcoming: return "Upcoming"
        case .dueToday: return "Due Today"
        case .overdue: return "Overdue"
        case .completed: return "Completed"
        }
    }

    var urgency: Int {
        switch self {
        case .overdue: return 0
        case .dueToday: return 1
        case .upcoming: return 2
        case .completed: return 3
        }
    }
}

struct Memory: Codable, Identifiable {
    let id: String
    var title: String
    var eventType: EventType
    var dateISO: String
    var confidence: Confidence
    var sourceText: String
    var createdAtISO: String
    var completedAtISO: String?
    var notificationId: String?
}

struct MemoryCandidate {
    var title: String
    var eventType: EventType
    var dateISO: String
    var confidence: Confidence
    var sourceText: String
}
